### PR TITLE
 Show hotspot icon and exon info on mutation

### PR DESCRIPTION
**Pull request by zhx828**
This fixes https://github.com/oncokb/oncokb-pipeline/issues/283 and https://github.com/oncokb/oncokb/issues/3694

### CODE REVIEW

File: src/main/java/org/mskcc/oncokb/curation/repository/AlterationRepository.java [hunk 1]
```diff
@@ -42,12 +45,27 @@ public interface AlterationRepository extends JpaRepository<Alteration, Long>, J
...
         " (alteration.name =:query or alteration.alteration=:query )"
     )
     List<Alteration> findByNameOrAlterationAndGenesId(@Param("query") String query, @Param("id") Long geneId);
 
+    @Query(
+        "select distinct alteration from Alteration alteration" +
+        " left join fetch alteration.genes g" +
+        " left join fetch alteration.flags" +
+        " where g.id in :genes and " +
+        " alteration.consequence = :consequence and ((alteration.start <= :end and alteration.end >= :start) or (alteration.end >= :start and alteration.start <= :end)) "
```
Comment (calvinlu3) [inline src/main/java/org/mskcc/oncokb/curation/repository/AlterationRepository.java:60]:
```suggestion
        " alteration.consequence = :consequence and (alteration.start <= :end and alteration.end >= :start) "
```

This should be sufficient to check if two ranges overlap. The original had two duplicated conditioned with OR.

**Review by calvinlu3 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mskcc/oncokb/curation/domain/enumeration/HotspotFlagEnum.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.mskcc.oncokb.curation.domain.enumeration;
+
+/**
+ * The TranscriptFlagEnum enumeration.
+ */
+public enum HotspotFlagEnum {
+    HOTSPOT_V1,
```
[the remark this answers is not in the document]
Comment (zhx828) [reply src/main/java/org/mskcc/oncokb/curation/domain/enumeration/HotspotFlagEnum.java:7]: @calvinlu3 there is a V2 about to come out. I'm preping for that.

File: src/main/webapp/app/pages/curation/mutation/MutationsSection.tsx [hunk 3]
```diff
@@ -12,25 +12,47 @@ import { componentInject } from 'app/shared/util/typed-inject';
...
   const [openMutationCollapsibleIndex, setOpenMutationCollapsibleIndex] = useState<number>(null);
 
   const mutationSectionRef = useRef<HTMLDivElement>(null);
   const mutationScrollContainerRef = useRef<HTMLDivElement>(null);
 
+  useEffect(() => {
+    onValue(
+      ref(firebaseDb, `${getFirebaseGenePath(isGermline, hugoSymbol)}/mutations`),
+      snapshot => {
+        annotatedAltsCache.fetch(hugoSymbol, snapshot.val());
```
Comment (calvinlu3) [inline src/main/webapp/app/pages/curation/mutation/MutationsSection.tsx:50]: `fetch` takes `hugoSymbol` and `MutationQuery[]`, but `snapshot.val()` here is actually of type `Mutation[]`.

File: src/main/webapp/app/pages/curation/mutation/MutationsSection.tsx [hunk 4]
```diff
@@ -12,25 +12,55 @@ import { componentInject } from 'app/shared/util/typed-inject';
...
+            return {
+              name: mut.name,
+              alterations: mut.alterations,
+            };
+          })
+        );
+      },
+      { onlyOnce: true }
+    );
+  }, []);
```
Comment (bprize15) [inline src/main/webapp/app/pages/curation/mutation/MutationsSection.tsx:62]: Probably safer to add dependencies to dependency array

File: src/main/webapp/app/shared/links/Linkout.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React, { CSSProperties } from 'react';
+
+export const Linkout: React.FunctionComponent<{
+  to: string;
+  className?: string;
+  style?: CSSProperties;
+  addHttpsProtocol?: boolean;
+}> = props => {
+  let updatedLink = props.to;
+  if (props.addHttpsProtocol === undefined || props.addHttpsProtocol) {
```
Comment (bprize15) [inline src/main/webapp/app/shared/links/Linkout.tsx:10]: Are we defaulting to true here? In that case, think we should use default assignment

**Pull request merged**